File: ios/Plugin/WechatPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(WechatPlugin, "Wechat",
           CAP_PLUGIN_METHOD(echo, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(registerApp, CAPPluginReturnNone);
           CAP_PLUGIN_METHOD(login, CAPPluginReturnNone);
           CAP_PLUGIN_METHOD(single, CAPPluginReturnNone);
)

File: ios/Plugin/WechatPlugin.swift
import Foundation
import Capacitor

public enum WechatPluginError: String, LocalizedError {
    case ConfigurationEmpty = "Configuration is empty, check again"
}

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(WechatPlugin)
public class WechatPlugin: CAPPlugin {
    private let implementation = Wechat()
        
    public override func load() {
        guard let wechatAppId = self.getConfigValue("appId") as? String else {
            print("config not found")
            return
        }
        guard let universalLink = self.getConfigValue("universalLink") as? String else {
            print("config not found")
            return
        }
        
        implementation.registerApp(wechatAppId, universalLink: universalLink)
        
//        UIApplication.shared.delegate.
    }
    
    @objc func login(_ call: CAPPluginCall) {
        implementation.auth()
    }

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": "yo"
        ])
    }
    
    @objc func single(_ call: CAPPluginCall) {
        print("123")
        call.resolve()
    }
}
